### PR TITLE
Rd 6312 missing exectuion tags

### PR DESCRIPTION
**Pull request by OrrLumigo**

### CODE REVIEW

File: src/reporter.ts [hunk 1]
```diff
@@ -131,7 +134,9 @@ export const forgeAndScrubRequestBody = (spans, maxSendBytes): string | undefine
     logger.debug(`Trimmed spans due to size`);
   }
   logger.debug(
-    `Filtered [${spans.length - spans.length}] spans out, Took: [${new Date().getTime() - start}ms]`
+    `Filtered [${beforeLength - spans.length}] and scrubbed spans out, Took: [${
```
Comment (saartochner) [inline src/reporter.ts:137]: update comment

File: src/utils.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ export const OMITTING_KEYS_REGEXES = [
   'Signature',
   'Credential',
   'Authorization',
+  'authorization',
```
Comment (saartochner) [inline src/utils.ts:26]: There is an ignorecase, why did you add that?
image

File: src/reporter.test.ts [hunk 3]
```diff
@@ -589,6 +588,97 @@ describe('reporter', () => {
...
+      .mockImplementationOnce(() => {
+        throw new Error('Error');
+      });
+    const spans = [
+      { info: { httpInfo: { request: {}, response: {} } } },
+      { info: { httpInfo: { request: {}, response: {} } } },
+      { info: { httpInfo: { request: {}, response: {} } } },
+    ];
+    scrubSpans(spans);
+    expect(spans.length).toEqual(spans.length);
```
Comment (saartochner) [inline src/reporter.test.ts:679]: This line doesn't check anything 😂

File: src/reporter.test.ts [hunk 4]
```diff
@@ -589,6 +588,97 @@ describe('reporter', () => {
...
+      .mockReturnValueOnce(true)
+      .mockImplementationOnce(() => {
+        throw new Error('Error');
+      });
+    const spans = [
+      { info: { httpInfo: { request: {}, response: {} } } },
+      { info: { httpInfo: { request: {}, response: {} } } },
+      { info: { httpInfo: { request: {}, response: {} } } },
+    ];
+    scrubSpans(spans);
```
Comment (saartochner) [inline src/reporter.test.ts:678]: If the scrub failed in the third span, what do you send there?
can you check the return value?

File: src/reporter.test.ts [hunk 5]
```diff
@@ -589,6 +588,100 @@ describe('reporter', () => {
...
+      });
+    const beforeScrub = [
+      { info: { httpInfo: { request: { host: 'StepFunction' }, response: {} } } },
+      { info: { httpInfo: { request: {}, response: { body: 'body' } } } },
+      { info: { httpInfo: { request: {}, response: {} } } },
+    ];
+    const spans = [...beforeScrub];
+    scrubSpans(spans);
+
+    expect(spans).toEqual(beforeScrub);
```
Comment (saartochner) [inline src/reporter.test.ts:681]: I don't get it - if you fail to scrub, then you just send it like that?
I think that we better remove it

File: src/reporter.test.ts [hunk 6]
```diff
@@ -589,19 +588,161 @@ describe('reporter', () => {
...
+            transactionId: 'transactionId',
+          },
+          httpInfo: { host: 'StepFunction' },
+          resourceName: 'StepFunction',
+          messageId: 'messageId',
+        },
+      },
+    ];
+    scrubSpans(spans);
+    expect(spans.length).toEqual(spans.length);
```
Comment (saartochner) [inline src/reporter.test.ts:662]: ?

File: src/reporter.test.ts [hunk 7]
```diff
@@ -589,19 +588,161 @@ describe('reporter', () => {
...
+        info: {
+          httpInfo: {
+            request: {},
+            response: {
+              body: '"body"',
+            },
+          },
+        },
+      },
+    ]);
```
Comment (saartochner) [inline src/reporter.test.ts:730]: don't you prefer to write
`expect(spans).toEqual([{...}, beforeScrub[1]])`

File: src/reporter.test.ts [hunk 8]
```diff
@@ -589,19 +588,162 @@ describe('reporter', () => {
...
-    expect(logs[1].msg).toEqual('#LUMIGO# - WARNING - "Error in Lumigo tracer"');
-    expect(JSON.parse(logs[0].obj).message).toEqual('resultSpans.forEach is not a function');
-    expect(JSON.parse(logs[1].obj).message).toEqual('spans.map is not a function');
-    expect(JSON.parse(logs[0].obj).stack).toBeTruthy();
-    expect(JSON.parse(logs[1].obj).stack).toBeTruthy();
+    expect(logs.length).toEqual(3);
+    expect(logs[0].msg).toEqual('#LUMIGO# - DEBUG - "Filtered [NaN] spans out"');
+    expect(logs[1].msg.startsWith('#LUMIGO# - DEBUG - "Filtering and scrubbing, Took: [')).toEqual(
+      true
+    );
```
Comment (saartochner) [inline src/reporter.test.ts:745]: toBeTruthy

File: src/reporter.ts [hunk 9]
```diff
@@ -69,50 +69,65 @@ function scrub(payload: any, headers: any, sizeLimit: number, truncated = false)
...
+    to = 0;
+  while (from < resultSpans.length) {
+    if (safeExecute(scrubSpan, 'Failed to scrub span')(resultSpans[from])) {
+      resultSpans[to] = resultSpans[from];
+      to++;
     }
-  });
+    from++;
+  }
+  resultSpans.length = to;
```
Comment (saartochner) [inline src/reporter.ts:124]: use plain filters - this is after the trimming, I.e. max 1MB memory. This code seems dangerous.

**Review by saartochner — APPROVED** [review]
Great work!

**Review by doriaviram — APPROVED** [review]
